File: widgets/android/LyricsModels.kt
package com.example.dailylyrics

import kotlinx.serialization.Serializable

/**
 * API 응답 모델
 */
@Serializable
data class LyricsResponse(
    val success: Boolean,
    val data: LyricsData? = null,
    val error: String? = null
)

/**
 * 가사 데이터
 */
@Serializable
data class LyricsData(
    val lines: List<String>,
    val title: String,
    val album: String,
    val year: Int,
    val artist: String,
    val timestamp: String,
    val interval: String? = null,
    val albumFolder: String? = null
) {
    /**
     * 앨범 커버 URL 생성
     */
    fun getCoverImageURL(baseURL: String = "http://10.0.2.2:58384"): String? {
        if (albumFolder.isNullOrEmpty()) return null

        // URL 인코딩 (세미콜론 등 특수문자 처리)
        val encodedFilename = java.net.URLEncoder.encode("$albumFolder.webp", "UTF-8")
        return "$baseURL/covers/$encodedFilename"
    }
}

File: widgets/windows/LyricsModels.cs
using System.Text.Json.Serialization;

namespace DailyLyricsWidget
{
    /// <summary>
    /// API 응답 모델
    /// </summary>
    public class LyricsResponse
    {
        [JsonPropertyName("success")]
        public bool Success { get; set; }

        [JsonPropertyName("data")]
        public LyricsData? Data { get; set; }

        [JsonPropertyName("error")]
        public string? Error { get; set; }
    }

    /// <summary>
    /// 가사 데이터
    /// </summary>
    public class LyricsData
    {
        [JsonPropertyName("lines")]
        public List<string> Lines { get; set; } = new();

        [JsonPropertyName("title")]
        public string Title { get; set; } = string.Empty;

        [JsonPropertyName("album")]
        public string Album { get; set; } = string.Empty;

        [JsonPropertyName("year")]
        public int Year { get; set; }

        [JsonPropertyName("artist")]
        public string Artist { get; set; } = string.Empty;

        [JsonPropertyName("timestamp")]
        public string Timestamp { get; set; } = string.Empty;

        [JsonPropertyName("interval")]
        public string? Interval { get; set; }

        [JsonPropertyName("albumFolder")]
        public string? AlbumFolder { get; set; }

        /// <summary>
        /// 앨범 커버 URL 생성
        /// </summary>
        public string? GetCoverImageURL(string baseURL = "http://127.0.0.1:58384")
        {
            if (string.IsNullOrEmpty(AlbumFolder))
                return null;

            // URL 인코딩 (세미콜론 등 특수문자 처리)
            var encodedFilename = Uri.EscapeDataString($"{AlbumFolder}.webp");
            return $"{baseURL}/covers/{encodedFilename}";
        }
    }
}
